palantir-java-format/build.gradle

```diff
@@ -30,40 +30,28 @@ dependencies {
     annotationProcessor 'org.derive4j:derive4j'
 }
 
-def exports = [
-        'jdk.compiler/com.sun.tools.javac.file',
-        'jdk.compiler/com.sun.tools.javac.main',
-        'jdk.compiler/com.sun.tools.javac.parser',
-        'jdk.compiler/com.sun.tools.javac.tree',
-        'jdk.compiler/com.sun.tools.javac.util',
-        'jdk.compiler/com.sun.tools.javac.code',
-        'jdk.compiler/com.sun.tools.javac.api'
-]
-
-def jvmArgList = exports.collect { value -> "--add-exports=${value}=ALL-UNNAMED".toString() }
-
-tasks.withType(JavaCompile).configureEach {
-    options.errorprone.disable 'StrictUnusedVariable'
-
-    // Allow access to internal javac apis
-    options.compilerArgs += jvmArgList
-
-    if (JavaVersion.current() < JavaVersion.VERSION_14) {
-        excludes = ['**/Java14InputAstVisitor.java']
-    }
+moduleJvmArgs {
+    exports().addAll(
+            'jdk.compiler/com.sun.tools.javac.file',
+            'jdk.compiler/com.sun.tools.javac.main',
+            'jdk.compiler/com.sun.tools.javac.parser',
+            'jdk.compiler/com.sun.tools.javac.tree',
+            'jdk.compiler/com.sun.tools.javac.util',
+            'jdk.compiler/com.sun.tools.javac.code',
+            'jdk.compiler/com.sun.tools.javac.api')
 }
 
-tasks.withType(Test).configureEach {
-    jvmArgs = jvmArgList
+tasks.withType(JavaCompile).named('compileJava') {
```

Comment:
By doing this just for `compileJava` we can use higher level language features in the tests